patternfly/patternfly-elements
# DE21128-broadcast-theme-hooks

* Link(s) to demo pages where this element can be viewed:
http://localhost:8000/elements/pfe-card/demo/

### What has changed and why
- Added support for manual overrides to which theme context is invoked by surface colors. Previously the theme was "locked" into having 3 light theme colors, 2 dark, and 2 saturated. Now the themes can be redefined if desired.
- Added demo file for pfe-card to test.

### Testing instructions
Be sure to include detailed instructions on how your update can be tested by another developer.

1. Open the elements/pfe-card/demo/custom-theme.css
2. Change the values of the following variables. Make stuff up, but make the colors match the appropriate theme, like this: 
```css
  --pfe-theme--color--surface--lightest: purple;
  --pfe-theme--color--surface--lightest--theme: dark;
  --pfe-theme--color--surface--lighter: maroon;
  --pfe-theme--color--surface--lighter--theme: dark;
  --pfe-theme--color--surface--base: green;
  --pfe-theme--color--surface--base--theme: dark;
  --pfe-theme--color--surface--darker: pink;
  --pfe-theme--color--surface--darker--theme: light;
  --pfe-theme--color--surface--darkest: yellow;
  --pfe-theme--color--surface--darkest--theme: light;
  --pfe-theme--color--surface--complement: hotpink;
  --pfe-theme--color--surface--complement--theme: saturated;
  --pfe-theme--color--surface--accent: #c0ff33;
  --pfe-theme--color--surface--accent--theme: light;
```
3. The cards should reflect both the color and theme changes. 

![image](https://user-images.githubusercontent.com/456863/69677416-eace6d80-1070-11ea-9170-00ed15b72473.png)

#### Browser requirements
Your component should work in all of the following environments:


### Ready-for-merge Checklist
